### DISCUSSION TITLE
'db error: ERROR: invalid input syntax for type timestamp: "$1"'

### DISCUSSION THREAD

**Discussion by leo-petrucci** [prisma/prisma#4347] [Q&A] [answered]
I'm having some trouble using `$queryRaw` together with variables.

This is my query:

```
        const thisMonth: string = '2020-11-01T00:00:00'
        const servers = ctx.prisma
          .$queryRaw`SELECT s.id, s.title, s.content
          FROM "Server" AS s 
          WHERE s."createdAt" >= '${thisMonth}';`
```

For whatever reason, trying to run this will return a `'db error: ERROR: invalid input syntax for type timestamp: "$1"'` error. But replacing the variable with a simple string (`WHERE s."createdAt" >= '2020-11-01T00:00:00';`) will run without errors.

I've used variables in queryRaw before and never had problems, it just seems specific to dates.

Currently using Postgres as a database.

Anyone know why it's behaving like this?

**Comment by leo-petrucci** [comment, marked as answer]
I tried changing my variable to a Javascript date object `const thisMonth: Date = new Date(`2020-11-01`)` and that just ended up working.

This should probably be in the docs somewhere? Unless I missed it?

**Reply by ryands17** [reply]
@creativiii The field should normally accept ISO strings, so `new Date().toISOString()` should also work.

**Reply by leo-petrucci** [reply]
That doesn't work, as previously mentioned. Only Date objects seem to work properly.

**Reply by tlantli** [reply]
Does this still work for you?

**Reply by stagefright5** [reply]
Be sure to also remove quotes ('') that surrounds the variable. (as mentioned here: https://github.com/prisma/prisma/discussions/4347#discussioncomment-2756954)

**Reply by bigbluespace** [reply]
Only date object and removing '' worked for me.

**Comment by C0rn3j** [comment]
This is because you are using a template literal (``) but also single-quoting the variable on top `'${thisMonth}'`. Remove the single quotes and it should work.

```javascript
const servers = ctx.prisma
          .$queryRaw`SELECT s.id, s.title, s.content
          FROM "Server" AS s 
          WHERE s."createdAt" >= ${thisMonth};`
```

**Reply by leo-petrucci** [reply]
This was a bug that was fixed last month afaik, but thank you for the answer!

**Reply by rodolphoasb** [reply]
Hey, @creativiii, do you have the issue number, please?

**Reply by leo-petrucci** [reply]
@rodolphoasb Sorry I had it confused with another issue 😅

It's definitely not related to the quotes though, I'm sure I had tried it both ways.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
